code/datums/abilities/flock/flockmind.dm

```diff
@@ -390,21 +388,33 @@
 	if(locate(/obj/flock_structure) in T)
 		boutput(holder.owner, "<span class='alert'>There is already a flock structure on this flocktile!</span>")
 		return 1
+
 	for (var/atom/O in T.contents)
 		if (O.density && !isflock(O))
 			boutput(holder.owner, "<span class='alert'>That tile has something that blocks tealprint creation!</span>")
 			return 1
+
+	var/list/friendlyNames = list()
+	var/mob/living/intangible/flock/flockmind/F = holder.owner
+	for(var/datum/unlockable_flock_structure/ufs as anything in F.flock.unlockableStructures)
+		if(ufs.check_unlocked())
```

Comment:
ufs has `ufs.unlocked` doesn't it?

Reply:
unlocked is updated every processing cycle, check_unlocked() gets a current value. If I just get unlocked, there's a ~3 second window where you could potentially build a structure you shouldn't be able to. For the cost of a single function call, it's worth the check.